MachObfuscator.xcodeproj/project.pbxproj

```diff
@@ -266,6 +266,9 @@
 		D4E365FA227CF5DE00FF7B9A /* SimpleSourceSymbolsLoader_loadFromFrameworkURL_allSystemFrameworks_Tests.swift in Sources */ = {isa = PBXBuildFile; fileRef = D4E365F9227CF5DE00FF7B9A /* SimpleSourceSymbolsLoader_loadFromFrameworkURL_allSystemFrameworks_Tests.swift */; };
 		D4E7634F201CDB60004300EB /* NibArchive+Loading.swift in Sources */ = {isa = PBXBuildFile; fileRef = D4E7634E201CDB60004300EB /* NibArchive+Loading.swift */; };
 		D4E76351201CDF05004300EB /* NibArchive+Nib.swift in Sources */ = {isa = PBXBuildFile; fileRef = D4E76350201CDF05004300EB /* NibArchive+Nib.swift */; };
+		D4EEC7D1236B3B58008D0E8A /* ObjectSymbols+blacklist.swift in Sources */ = {isa = PBXBuildFile; fileRef = D4EEC7D0236B3B58008D0E8A /* ObjectSymbols+blacklist.swift */; };
```

Comment:
you usually name extensions starting with capital letter: `+blacklist` -> `+Blacklist`

Reply:
Yes, it is a Obj-C habit (Obj-C category always had a name). Sometimes I create an extension with only one method, then I usually name the class with the method name (`TypeName+methodName`). Do you think it is a bad practice?

Reply:
I was not aware that this was your convention with capitalisation but it is indeed used in the sources. I see one problem with it: sometimes such small file may grow and start to contain more than that one method - should you rename it then? One method per file probably also is not a best general guideline.
I think that this convention does not solve many problem and brings confusion and inconsistencies.

Reply:
> sometimes such small file may grow and start to contain more than that one method - should you rename it then?

As long as the extension provides only one public/internal method, I think it is fine. This single method is the only thing that matters outside this file. All other private methods are just an implementation detail. I don’t want to create unit tests for all the helper methods, instead I would like to use the file as a blackbox. If the methods got to large/complicated, I would just refactor it, possibly adding unit tests for unprivatized code and possibly changing the file name.

> One method per file probably also is not a best general guideline.

Why do you think so? Should I move this code to some existing class and remove this unit test?

> brings confusion and inconsistencies

Is the confusion long lasting? I mean the problem with `Type+Suffix` naming is it is Obj-C originated, so this is kind of legacy convention (suffix being a category name). I don’t always name the file exactly as the types inside. E.g. I have a `Networking.swift` in my side project that provides my own nano-framework and it contains few types (in that particular example it just feels better to have it in one file, the file is about the screen size). Should I made the suffix name up, when the method name describes the contents best? Do you think we should stick to that Obj-C convention after all?

I don’t want to force this convention, but it will make me hard to apply something I do not believe or feel :) Please try to convince me.

Reply:
> > One method per file probably also is not a best general guideline.
> 
> Why do you think so? Should I move this code to some existing class and remove this unit test?

I think better guideline is: put related code together and unrelated code apart. Number-based guidelines IMO are too strict.
 
> > brings confusion and inconsistencies
> 
> Is the confusion long lasting? 

Some files are named like `ObjectSymbols+blacklist.swift`, others like `NibArchive+Loading.swift`. This is inconsistent for me, at least at first sight. You explained it, but it is not obvious. I think `ObjectSymbols+Blacklist.swift` would be perfectly fine.

> I mean the problem with `Type+Suffix` naming is it is Obj-C originated, so this is kind of legacy convention (suffix being a category name). I don’t always name the file exactly as the types inside. E.g. I have a `Networking.swift` in my side project that provides my own nano-framework and it contains few types (in that particular example it just feels better to have it in one file, the file is about the screen size). Should I made the suffix name up, when the method name describes the contents best? Do you think we should stick to that Obj-C convention after all?

For me, purpose of Category or extension is the same: to add group of related features/methods/properties etc. I see no difference between ObjC categories and Swift extensions from this point of view (but im neither ObjC nor Swift pro).

If method name is good (because it describes purpose/concept) and the concept requires only one public method to implement (IMO this is not always possible) then public method name may be the same as concept name.

Lets see how it works in this PR: you add a "concept" of "Blacklisting" to `ObjectSymbols`. This "concept" is implemented using `blacklist` method and currently does not need more but blacklisting sometimes involves also whitelisting. I think that naming file using "concept" is at higher level of abstraction that naming it after method name. Appropriate level of abstraction is probably more important than being ready for future changes (which may or may not be needed - you can always say, that you will refactor when there is need for it - and you probably will be right).

Reply:
> > > One method per file probably also is not a best general guideline.
> > 
> > 
> > Why do you think so? Should I move this code to some existing class and remove this unit test?
> 
> I think better guideline is: put related code together and unrelated code apart. Number-based guidelines IMO are too strict.

I agree fully. :) That is what I have done.

> > > brings confusion and inconsistencies
> > 
> > 
> > Is the confusion long lasting?
> 
> Some files are named like `ObjectSymbols+blacklist.swift`, others like `NibArchive+Loading.swift`. This is inconsistent for me, at least at first sight. You explained it, but it is not obvious. I think `ObjectSymbols+Blacklist.swift` would be perfectly fine.
> 
> > I mean the problem with `Type+Suffix` naming is it is Obj-C originated, so this is kind of legacy convention (suffix being a category name). I don’t always name the file exactly as the types inside. E.g. I have a `Networking.swift` in my side project that provides my own nano-framework and it contains few types (in that particular example it just feels better to have it in one file, the file is about the screen size). Should I made the suffix name up, when the method name describes the contents best? Do you think we should stick to that Obj-C convention after all?
> 
> For me, purpose of Category or extension is the same: to add group of related features/methods/properties etc. I see no difference between ObjC categories and Swift extensions from this point of view (but im neither ObjC nor Swift pro).

Neither am I the ObC/Swift pro. :) Obviously a Swift extension is much powerful tool than ObjC category. But aside from that, Swift extension is more kind of anonymous category, it doesn't need to have a name, also the Swift itself is more friendly for putting multiple extension in a single file. This allows you to do a lot of different setups using extensions.

> If method name is good (because it describes purpose/concept) and the concept requires only one public method to implement (IMO this is not always possible) then public method name may be the same as concept name.
> 
> Lets see how it works in this PR: you add a "concept" of "Blacklisting" to `ObjectSymbols`. This "concept" is implemented using `blacklist` method and currently does not need more but blacklisting sometimes involves also whitelisting. I think that naming file using "concept" is at higher level of abstraction that naming it after method name. Appropriate level of abstraction is probably more important than being ready for future changes (which may or may not be needed - you can always say, that you will refactor when there is need for it - and you probably will be right).

That is logical. Avoiding confusion – this is the nice argument in favour of changing my naming habits. I will change the file name and I will try to change my habbits. :) Thanks for interesting discussion!